crates/ra_proc_macro_srv/src/cli.rs

```diff
@@ -8,8 +8,9 @@ pub fn run() {
     loop {
         let req = match read_request() {
             Err(err) => {
-                eprintln!("Read message error on ra_proc_macro_srv: {}", err);
-                continue;
+                // Panic here, as the stdin pipe may be closed.
+                // Otherwise, client will be restarted the service anyway.
+                panic!("Read message error on ra_proc_macro_srv: {}", err);
```

Comment:
Let's maybe just buble the error with `?` and return an `io::Error` from `run`? In general,  it's almost always better to let the caller to deal with errors.

Reply:
Fixed.